topi/tests/python/test_topi_math.py

```diff
@@ -71,7 +70,8 @@ def check_device(device):
     test_apply(topi.abs, "fabs", np.abs, -100, 100)
     test_apply(topi.round, "round", np.round, -100, 100, check_round=True)
     test_apply(topi.exp, "exp", np.exp, -1, 1)
-    test_apply(topi.tanh, "tanh", np.tanh, -10, 10)
+    test_apply(topi.tanh, "tanh", np.tanh, -20, 20)
```

Comment:
Maybe also add a test that compares topi.tanh and np.tanh on 10^5 random inputs with [`np.testing.assert_array_almost_equal_nulp`](https://docs.scipy.org/doc/numpy/reference/generated/numpy.testing.assert_array_almost_equal_nulp.html) and verifies the diff is within a few ulp (as the impl claims it is?).

Reply:
I replaced the original `tvm.testing.assert_allclose` with `np.testing.assert_array_max_ulp`.
The float32 implementation is within 1 ulp (0.5 ulp is too small). The double precision one (with the GLIBC implementation) is within 3 ulps.